From 9410ad8202d67248a42bdc87cda87b80239e172b Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 15 Feb 2024 13:39:49 +0100
Subject: [PATCH 0594/1695] dts: stm32: Enable PWR IRQs for stm32mp135f-dk

Enable Wake-up pins (PWR IRQs) that is used by the PMIC driver.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 05d4a0043572a088c0de7cfeac6a74d9d080ef09 dts: stm32: Enable PWR IRQs for stm32mp135f-dk]
Change-Id: I51c24b7d96260912432d571475853190ec7479d2
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts | 25 ++++++++++++++++++++++++-
 1 file changed, 24 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index 21433fb0f..32d7c6ab4 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -9,6 +9,7 @@
 #include <dt-bindings/clock/stm32mp13-clksrc.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/gpio/stm32mp_gpio.h>
+#include <dt-bindings/mfd/st,stpmic1.h>
 #include <dt-bindings/power/stm32mp1-power.h>
 #include <dt-bindings/regulator/st,stm32mp13-regulator.h>
 #include <dt-bindings/soc/stm32mp-tzc400-macro.h>
@@ -33,6 +34,13 @@
 		stdout-path = "serial0:115200n8";
 	};
 
+	magic_wol: magic_wol {
+		compatible = "st,stm32mp1-pwr-irq-user";
+		wakeup-parent = <&pwr_irq>;
+		st,wakeup-pin-number = <2>;
+		status = "okay";
+	};
+
 	memory@c0000000 {
 		device_type = "memory";
 		reg = <0xc0000000 0x20000000>;
@@ -50,6 +58,14 @@
 		};
 	};
 
+	typec_wakeup: typec_wakeup {
+		compatible = "st,stm32mp1-pwr-irq-user";
+		wakeup-parent = <&pwr_irq>;
+		st,wakeup-pin-number = <5>;
+		st,notif-it-id = <1>;
+		status = "okay";
+	};
+
 	vin: vin {
 		compatible = "regulator-fixed";
 		regulator-name = "vin";
@@ -159,8 +175,10 @@
 		compatible = "st,stpmic1";
 		reg = <0x33>;
 		status = "okay";
+		wakeup-parent = <&pwr_irq>;
 		st,wakeup-pin-number = <1>;
-		st,notif-it-id = <0>;
+		st,pmic-it-id = <IT_PONKEY_F IT_PONKEY_R>;
+		st,notif-it-id = <0 2>;
 
 		regulators {
 			compatible = "st,stpmic1-regulators";
@@ -338,6 +356,11 @@
 	st,non-secure-otp-provisioning;
 };
 
+&pwr_irq {
+	status = "okay";
+	wakeup-gpios = <&gpiof 8 GPIO_ACTIVE_HIGH>, <&gpioi 3 GPIO_ACTIVE_HIGH>, <0>, <0>, <&gpioi 2 GPIO_PULL_UP>, <0>;
+};
+
 &pwr_regulators {
 	system_suspend_supported_soc_modes = <
 		STM32_PM_CSLEEP_RUN
-- 
2.47.2

